plugins/pulp_rpm/plugins/distributors/yum/configuration.py

```diff
@@ -461,13 +461,17 @@ def _check_for_relative_path_conflicts(repo, config, config_conduit, error_messa
     relative_path = get_repo_relative_path(repo, config)
     conflicting_distributors = config_conduit.get_repo_distributors_by_relative_url(relative_path,
                                                                                     repo.id)
-
     # in all honesty, this loop should execute at most once
     # but it may be interesting/useful for erroneous situations
     for distributor in conflicting_distributors:
         conflicting_repo_id = distributor['repo_id']
-        conflicting_relative_url = distributor['config']['relative_url'] or conflicting_repo_id
-        msg = _('Relative URL [%(p)s] for repository [%(r)s] conflicts with existing '
-                'relative URL [%(u)s] for repository [%(c)s]')
+        conflicting_relative_url = None
+        if 'relative_url' in distributor['config']:
+            conflicting_relative_url = distributor['config']['relative_url']
+            msg = _('Relative URL [%(p)s] for repository [%(r)s] conflicts with existing '
+                    'relative URL [%(u)s] for repository [%(c)s]')
+        else:
+            msg = _('Relative URL [%(p)s] for repository [%(r)s] conflicts with repo id '
+                    'for existing repository [%(c)s]')
         error_messages.append(msg % {'p': relative_path, 'r': repo.id,
```

Comment:
I would recommend using "".format() instead of % as the % operator is deprecated as of python 2.6.  See https://docs.python.org/2/library/stdtypes.html#str.format